src/pygeometadata/mcf.py

```diff
@@ -0,0 +1,413 @@
+import logging
+import os
+import yaml
+
+import jsonschema
+from jsonschema.exceptions import ValidationError
+import pygeometa.core
+from pygeometa.schemas.iso19139 import ISO19139OutputSchema
+from pygeometa.schemas.iso19139_2 import ISO19139_2OutputSchema
+import pygeoprocessing
+from osgeo import gdal
+from osgeo import ogr
+from osgeo import osr
+
+# Keep the yaml human-readable by avoiding anchors and aliases
+# https://stackoverflow.com/questions/13518819/avoid-references-in-pyyaml
+yaml.Dumper.ignore_aliases = lambda *args: True
+
+LOGGER = logging.getLogger(__name__)
+
+MCF_SCHEMA_FILE = os.path.join(
+    pygeometa.core.SCHEMAS, 'mcf', 'core.yaml')
+with open(MCF_SCHEMA_FILE, 'r') as schema_file:
+    MCF_SCHEMA = pygeometa.core.yaml_load(schema_file)
+
+# modify the core MCF schema so that our default
+# template MCFs have all the properties we expect
+# users to use.
+MCF_SCHEMA['required'].append('content_info')
+MCF_SCHEMA['properties']['content_info']['required'].append(
+    'attributes')
+MCF_SCHEMA['properties']['identification']['properties'][
+    'keywords']['patternProperties']['^.*'][
+    'required'] = ['keywords', 'keywords_type']
+
+# It's not clear to me why 'spatial' is type 'array' instead of
+# 'object', since it contains 'properties'.
+MCF_SCHEMA['properties']['identification']['properties'][
+    'extents']['properties']['spatial']['type'] = 'object'
```

Comment:
@phargogh I just discovered that overriding this bit of the MCF schema results in mcf docs that don't translate to XML using pygeometa's jinja templates. So I'll probably need to revert back to using `array` here and in a couple other places before we can write XML.

Reply:
Yes, what I don't understand (though it seems to work) is why the jsonschema for an array (`spatial`) has "properties" instead of "items". From pygeometa's `core.yml`:

```yaml
            extents:
                type: object
                description: spatial and temporal extents
                properties:
                    spatial:
                        type: array
                        description: spatial extent and CRS
                        properties:
                            bbox:
                                type: array
                                description: bounding box of resource
                                items:
                                    type: number
                                minItems: 4
                                maxItems: 6
                            crs:
                                type: string
                                description: coordinate reference system of bbox
                                default: 'http://www.opengis.net/def/crs/OGC/1.3/CRS84'
                            description:
                                type: string
                                description: description of the geographic area using an identifier
                        required:
                            - bbox
```

Reply:
Interesting!  I agree, that does seem like an oversight.  Looking at ISO19115, it appears that the extent is allowed to be defined N times, perhaps for N possible layers, in sorted order?  Even so, it appears that pygeometa's Jinja2 template only ever uses the first extent defined here, so maybe this isn't a commonly used part of the ISO spec.